admin/securedrop_admin/__init__.py

```diff
@@ -648,14 +646,33 @@ def update(args):
                                        keyserver=secondary_keyserver)
 
     git_verify_tag_cmd = ['git', 'tag', '-v', latest_tag]
-    sig_result = subprocess.check_output(git_verify_tag_cmd,
-                                         stderr=subprocess.STDOUT,
-                                         cwd=args.root)
+    try:
+        sig_result = subprocess.check_output(git_verify_tag_cmd,
+                                             stderr=subprocess.STDOUT,
+                                             cwd=args.root)
+
+        good_sig_text = 'Good signature from "SecureDrop Release Signing Key"'
+        bad_sig_text = 'BAD signature'
+        # To ensure that an adversary cannot name a malicious key good_sig_text
+        # we check that bad_sig_text does not appear.
+        if sig_result.count(RELEASE_KEY) == 1 and \
+                sig_result.count(good_sig_text) == 1 and \
+                bad_sig_text not in sig_result:
```

Comment:
Nit: since this is a very delicate logic, it would be nice to have test cases for all branches of the expression (i.e. one case where the name of the key is **Good signature from "SecureDrop Release Signing Key"**, another where the name of the key contains **22245C81E3BAEB4138B36061310F561200F4AD77**). Absolutely not a blocker for the merge though.